### PR TITLE
Windows: tell cmake it's not crosscompiling in MSYS2

### PR DESCRIPTION
**Pull request by iDunk5400**
This fixes a cmake error in MSYS2 introduced by #4294

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
While I know sfa about cmake/windows, this seems unconditional. Will this actually break cross compiling ?

**Comment by iDunk5400** [comment]
That's why I used the CMAKE_HOST_WIN32 check, so if host is not Windows it should still work.

**Comment by moneromooo-monero** [comment]
OK, but that doesn't mean the host is different from the target, does it ?

**Comment by iDunk5400** [comment]
How would you interpret this ?
```
CMake Warning at translations/CMakeLists.txt:35 (message):
  CrossCompiling


CMake Error at translations/CMakeLists.txt:37 (INCLUDE):
  INCLUDE could not find load file:

    C:/msys64/home/idunk/monero/build/release/translations/ImportExecutables.cmake
```
That's without the patch.

https://cmake.org/cmake/help/v3.0/variable/CMAKE_CROSSCOMPILING.html

**Comment by iDunk5400** [comment]
Hmm, another way of fixing this would be to remove [this](https://github.com/monero-project/monero/blob/master/cmake/64-bit-toolchain.cmake#L29) line. Then `CMAKE_SYSTEM` and `CMAKE_HOST_SYSTEM` would match, and `CMAKE_CROSSCOMPILING` would not be set.

Relevant cmake vars with L29:
```
-- CMAKE_HOST_SYSTEM=Windows-10.0.17134
-- CMAKE_HOST_SYSTEM_NAME=Windows
-- CMAKE_HOST_SYSTEM_PROCESSOR=AMD64
-- CMAKE_HOST_SYSTEM_VERSION=10.0.17134

-- CMAKE_SYSTEM=Windows
-- CMAKE_SYSTEM_NAME=Windows
-- CMAKE_SYSTEM_PROCESSOR=
-- CMAKE_SYSTEM_VERSION=
-- CMAKE_CROSSCOMPILING=TRUE
```
Relevant cmake vars without L29:
```
-- CMAKE_HOST_SYSTEM=Windows-10.0.17134
-- CMAKE_HOST_SYSTEM_NAME=Windows
-- CMAKE_HOST_SYSTEM_PROCESSOR=AMD64
-- CMAKE_HOST_SYSTEM_VERSION=10.0.17134

-- CMAKE_SYSTEM=Windows-10.0.17134
-- CMAKE_SYSTEM_NAME=Windows
-- CMAKE_SYSTEM_PROCESSOR=AMD64
-- CMAKE_SYSTEM_VERSION=10.0.17134
-- CMAKE_CROSSCOMPILING=FALSE
```

**Comment by MoroccanMalinois** [comment]
`grep -R CMAKE_CROSSCOMPILING Source` in cmake source code shows that it's only used to automatically skip execution of binaries, with [this](https://github.com/Kitware/CMake/blob/master/Source/cmFileCommand.cxx#L1142) exception, in a function named "SetPermissions" :

    #ifdef WIN32
      if (Makefile->IsOn("CMAKE_CROSSCOMPILING")) {
        // Store the mode in an NTFS alternate stream.
        std::string mode_t_adt_filename =
          std::string(toFile) + ":cmake_mode_t";

        // Writing to an NTFS alternate stream changes the modification
        // time, so we need to save and restore its original value.
        [...]

I understood it as a way to preserve file permissions when they will be moved to windows, but this could be completely wrong!

So, imho, `CMAKE_SYSTEM_NAME Windows` should only be set when using mingw on linux, target that doesn't exists in current Makefile (wondering if it just needs a different MSYS2_FOLDER ...)

tl;dr i vote for @iDunk5400 second solution
